exporter.go

```diff
@@ -63,75 +66,133 @@ func hashNameAndLabels(name string, labels prometheus.Labels) uint64 {
 	return hash.Sum64()
 }
 
+func labelsNames(labels prometheus.Labels) []string {
+	names := make([]string, 0, len(labels))
+	for labelName := range labels {
+		names = append(names, labelName)
+	}
+	sort.Strings(names)
+	return names
+}
+
+// hashNameAndLabels returns a hash value of the provided name string and all
+// the label names in the provided labels map.
+//
+// Not safe for concurrent use! (Uses a shared buffer and hasher to save on
+// allocations.)
```

Comment:
could they be taken out of a pool safely? would that be beneficial? what is the consequence of serialised access for ingestion and scrape performance?

Reply:
I don't fully get your question but has feeling that partly answered there: https://github.com/prometheus/statsd_exporter/pull/164#discussion_r235991822